neo/Wallets/Wallet.cs

```diff
@@ -318,20 +334,20 @@ private Transaction MakeTransaction(Snapshot snapshot, TransactionAttribute[] at
                             tx.SystemFee -= remainder;
                     }
                 }
-                UInt160[] hashes = tx.GetScriptHashesForVerifying(snapshot);
+                tx.Witnesses = null;
                 int size = Transaction.HeaderSize + attributes.GetVarSize() + script.GetVarSize() + IO.Helper.GetVarSize(hashes.Length);
                 foreach (UInt160 hash in hashes)
                 {
                     script = GetAccount(hash)?.Contract?.Script ?? snapshot.Contracts.TryGet(hash)?.Script;
                     if (script is null) continue;
                     if (script.IsSignatureContract())
                     {
-                        size += 66 + script.GetVarSize();
+                        size += 1 + 66 + script.GetVarSize();
                         tx.NetworkFee += ApplicationEngine.OpCodePrices[OpCode.PUSHBYTES64] + ApplicationEngine.OpCodePrices[OpCode.PUSHBYTES33] + InteropService.GetPrice(InteropService.Neo_Crypto_CheckSig, null);
                     }
                     else if (script.IsMultiSigContract(out int m, out int n))
                     {
-                        int size_inv = 65 * m;
+                        int size_inv = 1 + 65 * m;
                         size += IO.Helper.GetVarSize(size_inv) + size_inv + script.GetVarSize();
```

Comment:
A possible problem is that if we will sign the TX without Global scope, the size will be more than 1, 1+ScopedHash, so is hard to compute previous the signature

Reply:
This is the difference between the proposals Shargon. I'll fix this.

Reply:
Solved now, signature will already cover CosignerUsage (data on tx header). After that, we just need to test now, and ensure matching is happening between header data and witness data (perhaps it's not even needed on witness serialization).